fe/fe-core/src/main/java/com/starrocks/sql/common/PartitionDiffer.java

```diff
@@ -79,8 +79,8 @@ public static PartitionDiffer build(MaterializedView materializedView, TaskRunCo
         Range<PartitionKey> rangeToInclude = null;
         Column partitionColumn =
                 ((RangePartitionInfo) materializedView.getPartitionInfo()).getPartitionColumns().get(0);
-        String start = context.getProperties().get(TaskRun.PARTITION_START);
-        String end = context.getProperties().get(TaskRun.PARTITION_END);
+        String start = context == null ? null : context.getProperties().get(TaskRun.PARTITION_START);
+        String end = context == null ? null : context.getProperties().get(TaskRun.PARTITION_END);
```

Comment:
`context == null` means it's query and not refresh task?

Reply:
Yes, I want to reuse this function in loose mode query rewrite